### PR TITLE
NXP-23085: fix UserGroup.Suggestion with group restrictions

### PR DESCRIPTION
**Pull request by efge**
https://jira.nuxeo.com/browse/NXP-23085

### CODE REVIEW

**Comment by kevinleturc** [comment]
👍

**Comment by efge** [comment]
T&P in progress

File: nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/SuggestUserEntries.java [hunk 1]
```diff
@@ -245,30 +238,89 @@ public Blob run() throws IOException {
...
-                         return a;
-                     })
-                     .filter(result -> !result.isEmpty())
-                     // search on whole term to handle a whitespace within the first or last name
-                     .orElseGet(() -> userManager.searchUsers(trimmedPrefix));
+        String[] searchTerms = prefix.trim().split("\\s", 2);
+        List<Predicate> predicates = Arrays.stream(searchTerms)
+                                           .map(this::getUserSearchPredicate)
+                                           .collect(Collectors.toList());
+        DocumentModelList users = searchUsers(new MultiExpression(Operator.AND, predicates), limit);
```
Comment (ataillefer) [inline nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/SuggestUserEntries.java]: Could you please add back this comment here:
// intersection between all search results to handle full name

**Comment by ataillefer** [comment]
:+1:

File: nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/SuggestUserEntries.java (hunk 1 above)
Comment (efge) [reply nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/SuggestUserEntries.java]: Ok done.

**Pull request merged**